src/rws_client.cpp

```diff
@@ -147,10 +147,44 @@ RWSClient::RWSResult RWSClient::getMechanicalUnitJointTarget(const std::string&
   return evaluatePOCOResult(httpGet(uri), evaluation_conditions);
 }
 
-RWSClient::RWSResult RWSClient::getMechanicalUnitRobTarget(const std::string& mechunit)
+RWSClient::RWSResult RWSClient::getMechanicalUnitRobTarget(const std::string& mechunit,
+                                                           const Coordinate& coordinate,
+                                                           const std::string& tool,
+                                                           const std::string& wobj)
 {
   std::string uri = generateMechanicalUnitPath(mechunit) + Resources::ROBTARGET;
 
+  std::string args = "";
+  if (!tool.empty())
+  {
+    args += "&tool=" + tool;
+  }
+  if (!wobj.empty())
+  {
+    args += "&wobj=" + wobj;
+  }
+
+  const std::string coordinate_arg = "?coordinate=";
+  switch (coordinate)
```

Comment:
I would indent the `swich {...}` similar to [this](https://github.com/ros-industrial/abb_librws/blob/4741d83b865a1a495d9973294607d65c660d2f21/src/rws_poco_client.cpp#L98-L131), just to keep consistent code style.